Firmware/ultralcd.cpp

```diff
@@ -5106,30 +5106,24 @@ static void auto_deplete_switch()
     eeprom_update_byte((unsigned char *)EEPROM_AUTO_DEPLETE, lcd_autoDeplete);
 }
 
-static bool settingsAutoDeplete()
-{
-    if (mmu_enabled)
-    {
-        if (!fsensor_enabled)
-        {
-            if (menu_item_text_P(_i("SpoolJoin   [N/A]"))) return true;
-        }
-        else if (lcd_autoDeplete)
-        {
-            if (menu_item_function_P(_i("SpoolJoin    [on]"), auto_deplete_switch)) return true;
-        }
-        else
-        {
-            if (menu_item_function_P(_i("SpoolJoin   [off]"), auto_deplete_switch)) return true;
-        }
-    }
-    return false;
-}
-
 #define SETTINGS_AUTO_DEPLETE \
 do\
 {\
-    if(settingsAutoDeplete()) return;\
+	if (mmu_enabled)\
```

Comment:
These long macros may be hard to understand and debug (even the IDE's may have problems understanding this).
IMO the previous version with the settingsAutoDeplete() function call was easier to maintain.

Reply:
@DRracer I don't know how we could make it more readable. Do you mean the fact that it's all in the define with \ at the end? We could move it's contents to a static void function like before.
The thing is I did it the same on the Cutter.

Reply:
The static function like before is the preferred solution, thank you ;)